### PR TITLE
Issue #253: grade school improvements

### PR DESCRIPTION
**Pull request by davethomas11**

### CODE REVIEW

**Comment by matthewmorgan** [comment]
@davethomas11 looks like we have some merge conflicts due to merging previous PRs.  Would you mind resolving them and pushing again, or LMK and I can help with that.

**Comment by davethomas11** [comment]
@matthewmorgan Resolved the conflicts

File: exercises/grade-school/src/test/java/SchoolTest.java [hunk 1]
```diff
@@ -62,29 +62,27 @@ public void getsStudentsInEmptyGrade() {
...
+    sortedStudents.put(4, IsIterableContainingInOrder
+      .contains("Adam", "Christopher", "Jennifer", "Kyle", "Zed"));
+    sortedStudents.put(3, IsIterableContainingInOrder
+      .contains("Kylie"));
+
+    Map schoolStudents = school.studentsByGradeAlphabetical();
+    for (Map.Entry<?, Matcher> entry : sortedStudents.entrySet()) {
+
+      assertThat((Collection) schoolStudents.get(entry.getKey()), entry.getValue());
+    }
```
Comment (matthewmorgan) [inline exercises/grade-school/src/test/java/SchoolTest.java]: I like this!  Took me a minute to figure out what was going on, honestly.

**Comment by matthewmorgan** [comment]
@davethomas11 Thanks again!

**Pull request merged**